lib/nrf_modem_lib/custom_trace_medium/Kconfig

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+choice NRF_MODEM_LIB_TRACE_MEDIUM
+
+config NRF_MODEM_LIB_TRACE_MEDIUM_PRINTK
```

Comment:
I agree it would be nice to group the uart and rtt implementations in a folder, but I suggest to move the printk-sample to the `samples/nrf9160` folder to make it easier to find and make it clear that it's possible to create a custom trace medium in any folder.

Reply:
+1 for grouping, but would not move the printk medium. If it should be kept (is it useful? there are already two implementations) I would keep it together with the rest. I imagine it would be a little bit trickier to include it in the build from a sample folder.

Reply:
I think the printk medium isn't meant to be useful, but as an example on how to make your own custom trace medium. As far as I understand, this implementation is meant to be placed anywhere you want. So the application can have their custom implementation together with their application logic, and use the macro-functions to register the trace medium. It should be just the same to register the trace medium from a sample. To me it makes most sense to have examples in the samples folder.

Reply:
Yes. The idea is to have it as a reference on how to add mediums. Renamed the medium to `example` and placed it under the `samples/nrf9160/example_trace_medium` folder. If you can think of a better place, I will move it.

Reply:
There are many samples in that folder, and all of them are examples. I suggest renaming the `example_trace_medium` folder to `modem_trace_medium`.